src/api/routes/records.py

```diff
@@ -58,7 +127,31 @@ async def update_disciplinary_record(
     """
     logger.debug(f"{username} trying to update a DisciplinaryRecord")
     # TODO:
-    pass
+
+    permission_ok = Access.is_admin_write(username)
+    
+    if not permission_ok:
+        logger.debug(f"User({username}) permission denied.")
+        raise HTTPException(
+            status_code=401, detail="You don't have permission to update this."
+        )
+
+    try:
+        record_dict: dict = records_collection.find_one({'uid': uid})
+    except Exception as e:
+        logger.error("Failed to query database.")
+        logger.error(e)
+        raise HTTPException(status_code=500, detail="Database Error.")
+
+    record_dict = dict(record.dict())
+    remove_none_value_keys(record_dict)
+    try:
+        updated = students_collection.find_one_and_update(filter({"uid": uid}), update={"$set": record_dict})
+        return updated if updated else {"msg": "failed"}
```

Comment:
`clean_dict()` is required here, otherwise Pydantic will cause error for the key "_id" in the dictionary.